torchgeo/datasets/dior.py

```diff
@@ -0,0 +1,392 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""DIOR dataset."""
+
+import os
+from collections.abc import Callable
+from typing import Any, ClassVar
+from xml.etree import ElementTree
+
+import matplotlib.patches as patches
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import torch
+from matplotlib.figure import Figure
+from PIL import Image
+from torch import Tensor
+
+from .errors import DatasetNotFoundError
+from .geo import NonGeoDataset
+from .utils import (
+    Path,
+    check_integrity,
+    download_and_extract_archive,
+    download_url,
+    extract_archive,
+)
+
+
+def parse_pascal_voc(path: Path) -> dict[str, Any]:
+    """Read a PASCAL VOC annotation file.
+
+    Args:
+        path: path to xml file
+
+    Returns:
+        dict of image filename, bounding box coords, and class labels
+    """
+    et = ElementTree.parse(path)
+    element = et.getroot()
+    filename = element.find('filename').text  # type: ignore[union-attr]
+    labels, bboxes = [], []
+
+    for obj in element.findall('object'):
+        bndbox = obj.find('bndbox')
+        bbox = [
+            int(bndbox.find('xmin').text),  # type: ignore[union-attr, arg-type]
+            int(bndbox.find('ymin').text),  # type: ignore[union-attr, arg-type]
+            int(bndbox.find('xmax').text),  # type: ignore[union-attr, arg-type]
+            int(bndbox.find('ymax').text),  # type: ignore[union-attr, arg-type]
+        ]
+        label = obj.find('name').text  # type: ignore[union-attr]
+        bboxes.append(bbox)
+        labels.append(label)
+
+    return dict(filename=filename, bboxes=bboxes, labels=labels)
+
+
+class DIOR(NonGeoDataset):
+    """DIOR dataset.
+
+    `DIOR <https://arxiv.org/abs/1909.00133>`_ dataset contains horizontal bounding box
+    annotations of Google Earth Aerial RGB imagery. The test split does not contain bounding
+    box annotations and labels
+
+    Dataset features:
+
+    * 20 classes
+    * 192,472 manually annotated bounding box instances
+
+    Dataset format:
+
+    * Images are three channel .jpg files.
+    * Annotations are in `Pascal VOC XML format
+      <https://roboflow.com/formats/pascal-voc-xml>`_
+
+
+    Classes:
+
+    0. Airplane
+    1. Airport
+    2. Baseball Field
+    3. Basketball Court
+    4. Bridge
+    5. Chimney
+    6. Dam
+    7. Expressway Service Area
+    8. Expressway Toll Station
+    9. Golf Field
+    10. Ground Track Field
+    11. Harbor
+    12. Overpass
+    13. Ship
+    14. Stadium
+    15. Storage Tank
+    16. Tennis Court
+    17. Train Station
+    18. Vehicle
+    19. Windmill
+
+
+    If you use this dataset in your research, please cite the following paper:
+
+    * https://arxiv.org/abs/1909.00133
+
+
+    .. versionadded:: 0.7
+    """
+
+    url = 'https://huggingface.co/datasets/torchgeo/dior/resolve/main/{}'
```

Comment:
```suggestion
    url = 'https://hf.co/datasets/torchgeo/dior/resolve/ec7be9567d2e08eb3d3401c15a52ee2145d0ef01/{}'
```
Need a stable download URL